tests/generate-transaction-tests/index.js

```diff
@@ -271,11 +274,11 @@ const manifestTestnetPayloadCases = [].concat.apply([], testnetTemplates.map((te
     }), 
     TESTNET, 
   ])
-}));
+});
 
-const manifestMainnetPayloadCases = [].concat.apply([], mainnetTemplates.map((template) => {
+const manifestMainnetPayloadCases = mainnetTemplates.flatMap((template) => {
   const maxSv = numberOfRequiredTests(template.arguments);
-  return [...Array(maxSv).keys()].map((samplevalueNumber) => [
+  return (new Array(maxSv)).map((_,samplevalueNumber) => [
```

Comment:
this is actually a bad abstraction when you think about it - separating computation of required number of arguments from argument generation itself

I would prefer one function
`coverArgumentCombinations(template.arguments)`

and then use it as
```
let combinations = coverArgumentCombinations(template.arguments)
return combinations.map((args, i) => [
  combinations.length === 0 ? ...
  buildPayloadTx({script, args})
])
and use it as

Reply:
This is very good point. It seemed fishy to write such a long comment on argument combinations. After the refactor it should be obvious.